pandas/_testing.py

```diff
@@ -2724,11 +2724,10 @@ class for all warnings. To check that no warning is returned,
         extra_warnings = []
 
         for actual_warning in w:
```

Comment:
isn't this code functionally the same as existing?

Reply:
I thought if `expected_warning` is `False` or `None`, then `extra_warnings` does not get appended to because of the `continue`. So the check at the end for raising on extra warnings doesn't get triggered.

Reply:
Current implementation will ignore the else clause below, which consolidates extra warnings. I see that I messed that up in one of my previous PRs. Probably, if some functions are extracted, that would improve the readability of the logic (maybe in a separate PR).

Reply:
The added tests do raise on master as well, so behavior for them has changed